tests/units/anta_tests/test_docstrings.py

```
@@ -0,0 +1,58 @@
+@pytest.mark.parametrize("anta_test", ANTA_TESTS, ids=lambda test: f"{test.__module__}.{test.name}")
+def test_docstring_example(anta_test: type[AntaTest]) -> None:
+    """Verify that each test docstring contains a valid catalog for that test."""
+    example = extract_examples(anta_test.__doc__ or "")
+    assert example is not None, f"{anta_test.__module__}.{anta_test.name} is missing an Examples section"
+
+    match = re.fullmatch(r"```yaml\s*\n(?P<catalog>.*)\n\s*```", inspect.cleandoc(example), flags=re.DOTALL)
+    assert match is not None, f"{anta_test.__module__}.{anta_test.name} must contain one fenced YAML example"
+
+    catalog = AntaCatalogFile(yaml.safe_load(match.group("catalog")))
+    documented_tests = [test_definition.test for test_definitions in catalog.root.values() for test_definition in test_definitions]
+    assert documented_tests and all(documented_test is anta_test for documented_test in documented_tests)
```

Ruff (0.16.3)

[warning] Assertion should be broken down into multiple parts

Break down assertion into multiple parts

(PT018)